verification/cbmc/include/core_json_annex.h

```diff
@@ -67,45 +67,71 @@
  * Please see core_json.c for documentation.
  */
 
-void skipSpace( const char * buf,
-                size_t * start,
-                size_t max );
-
-bool skipUTF8( const char * buf,
-               size_t * start,
-               size_t max );
-
-bool skipEscape( const char * buf,
-                 size_t * start,
-                 size_t max );
-
-bool skipString( const char * buf,
-                 size_t * start,
-                 size_t max );
-
-bool skipAnyLiteral( const char * buf,
-                     size_t * start,
-                     size_t max );
-
-bool skipDigits( const char * buf,
-                 size_t * start,
-                 size_t max,
-                 int32_t * outValue );
+/* *INDENT-OFF* */
+JSONStatus_t JSON_Validate( const char * buf,
+                            size_t max )
+__CPROVER_requires( max < CBMC_MAX_BUFSIZE )
+__CPROVER_requires( buf == NULL || __CPROVER_is_fresh( buf, max ) )
+__CPROVER_ensures( jsonValidateEnum( __CPROVER_return_value ) );
 
-bool skipNumber( const char * buf,
-                 size_t * start,
-                 size_t max );
-
-bool skipSpaceAndComma( const char * buf,
-                        size_t * start,
-                        size_t max );
+JSONStatus_t skipCollection( const char * buf,
+                             size_t * start,
+                             size_t max )
+__CPROVER_requires( max > 0 && max < CBMC_MAX_BUFSIZE )
+__CPROVER_requires( __CPROVER_is_fresh( buf, max ) )
+__CPROVER_requires( __CPROVER_is_fresh( start, sizeof( *start ) ) )
+__CPROVER_assigns( *start )
+__CPROVER_ensures( skipCollectionEnum( __CPROVER_return_value ) )
+__CPROVER_ensures( __CPROVER_return_value == JSONSuccess ==> *start <= max );
```

Comment:
```suggestion
__CPROVER_ensures( (__CPROVER_return_value == JSONSuccess) ==> (*start <= max) );
```

Reply:
There are several of these cases where parentheses might make it more readable.  I've only marked a couple.

Reply:
I'll go over all the parenthesis and make sure to clean up our contracts.